R/format_KIL.R

```diff
@@ -0,0 +1,657 @@
+
+
+  #### Adults
+  #### Data from 1971 to 1992
+  capture_adults_til92 <-
+    adults_data_til92 %>%
+    dplyr::mutate(RingAge = "adult",
+                  Sex_observed = .data$Sex,
+                  BreedingSeason = as.integer(.data$RingDate),
+                  #### Create the date, as there is no information
+                  CaptureDate = as.Date(paste0(BreedingSeason, "-05-01")),
+                  CaptureTime = NA_character_,
+                  ObserverID  = NA_character_,
+                  CaptureAlive = TRUE,
+                  ReleaseAlive = .data$CaptureAlive,
+                  CapturePopID = .data$PopID,
+                  CapturePlot  = NA_character_,
+                  ReleasePopID = ifelse(ReleaseAlive == TRUE, CapturePopID, NA_character_),
+                  ReleasePlot  = ifelse(ReleaseAlive == TRUE, CapturePlot, NA_character_),
+                  Mass = NA_real_,
+                  Tarsus = NA_real_,
+                  #### ASK DATA OWNER
+                  OriginalTarsusMethod = NA_character_,
+                  #### ASK DATA OWNER what do the codes mean? Age or Euring codes?
+                  # Age_observed = case_when()
+                  ChickAge = NA_integer_,
+                  ExperimentID = NA_character_)
+
+
+
+  #### Chicks
+  #### Data from 1971 to 1992
+  capture_chicks_til92 <-
+    chicks_data_til92 %>%
+    dplyr::mutate(RingAge = "chick",
+                  Sex_observed = .data$Sex,
+                  BreedingSeason = as.integer(.data$RingDate),
+                  #### Create the date, as there is no information
+                  CaptureDate = as.Date(paste0(BreedingSeason, "-05-01")),
+                  CaptureTime = NA_character_,
+                  ObserverID  = NA_character_,
+                  CaptureAlive = TRUE,
+                  ReleaseAlive = .data$CaptureAlive,
+                  CapturePopID = .data$PopID,
+                  CapturePlot  = NA_character_,
+                  ReleasePopID = ifelse(ReleaseAlive == TRUE, CapturePopID, NA_character_),
+                  ReleasePlot  = ifelse(ReleaseAlive == TRUE, CapturePlot, NA_character_),
+                  Mass = .data$Mass,
+                  Tarsus = .data$Tarsus*0.777 + 6.158,
+                  OriginalTarsusMethod = "Standard",
+                  WingLength = NA_real_,
+                  Age_observed = 1L,
+                  ChickAge = 8L,
+                  ExperimentID = NA_character_)
+
+  #### Merge with brood data to get LocationID
+  capture_til92 <-
+    capture_adults_til92 %>%
+    dplyr::bind_rows(capture_chicks_til92) %>%
+    left_join(brood_data_til92_temp,
+              by = c("Species", "PopID", "BreedingSeason", "BroodId"))
+
+
+  #### Adults (breeding pairs)
+  #### Data after 1995
+  capture_adults_95 <-
+    adults_data_95 %>%
+    dplyr::mutate(RingAge = "adult",
+                  #### ASK DATA OWNER
+                  CaptureDate = NA_character_,
```

Comment:
This should still be a date object even if NA. This causes failure otherwise because we can't easily row bind character and date data.

```suggestion
                  CaptureDate = as.Date(NA),
```

Reply:
Ok, actually we now have the capture date in the updated dataset they sent.